crates/api_models/src/customers.rs

```diff
@@ -1,8 +1,52 @@
 use common_utils::{consts, crypto, custom_serde, pii};
-use masking::Secret;
+use masking::{Secret, SerializableSecret};
 use serde::{Deserialize, Serialize};
 use utoipa::ToSchema;
 
+use crate::enums as api_enums;
+
+impl SerializableSecret for AddressDetails {}
+
+#[derive(Debug, Default, Clone, Deserialize, Serialize, ToSchema)]
+#[serde(deny_unknown_fields)]
+pub struct AddressDetails {
```

Comment:
what was the requirement of creating a new struct `AddressDetails`.

Is this any different from `crates/api_models/src/payments.rs:1134`

Reply:
Nope, it's the same. But earlier 'customers` endpoint was taking `address` as `SecretSerdeValue` which was incorrect. So I just added this `AddressDetails` struct to take in only the specified values while making a request; the same way it's done for payments.

Reply:
If it is the same as that in the payments.rs file, then don't create a duplicate struct, the same struct can be imported and used here.

Reply:
Okay.